Soccar [Game-Server]/src/nl/soccar/gameserver/model/session/RoomWrapper.java

```diff
@@ -0,0 +1,261 @@
+        synchronized (players) {
+            if (players.stream()
+                    .map(PlayerWrapper::unwrap)
+                    .anyMatch(p -> p.equals(unwrapped))) {
+                player.getConnection().send(new JoinSessionMessage(JoinSessionMessage.Status.USERNAME_EXISTS));
+                return;
+            }
+
+            players.add(player);
+
+            unwrapped.setPlayerId(RoomUtilities.getNextPlayerId(this));
+            unwrapped.setCurrentSession(session.unwrap());
+        }
+
+        joinRandomTeam(player);
+        sendRoomInformation(player);
+
+        String roomName = room.getName();
+        LOGGER.log(Level.INFO, "Player {0} joined session {1}.", new String[]{player.getUsername(), roomName});
+
+        GameServerRmiController rmiController = GameServer.getInstance().getRmiControler();
+        rmiController.increaseOccupancy(roomName);
+        rmiController.changeHost(roomName, room.getHost().getUsername());
+    }
+
+    public void leave(PlayerWrapper player) {
+        synchronized (players) {
+            PlayerLeftSessionMessage message = new PlayerLeftSessionMessage(player.getPlayerId());
+
+            players.remove(player);
+            players.stream()
+                    .map(PlayerWrapper::getConnection)
+                    .forEach(c -> c.send(message));
+        }
+
+        Player unwrapped = player.unwrap();
+        unwrapped.setCurrentSession(null);
+        unwrapped.setPlayerId(0);
+
+        synchronized (teams) {
+            Team team = teams.remove(player);
+            if (team != null) {
+                team.leave(unwrapped);
+            }
+        }
+
+        String roomName = room.getName();
+        LOGGER.log(Level.INFO, "Player {0} left session {1}.", new String[]{player.getUsername(), roomName});
+
+        if (room.getOccupancy() <= 0) {
+            GameServer.getInstance().getSessionController().destroySession(room.getName());
+            return;
+        }
+
+        GameServerRmiController rmiController = GameServer.getInstance().getRmiControler();
+        rmiController.decreaseOccupancy(roomName);
+        rmiController.changeHost(roomName, room.getHost().getUsername());
+    }
+
+    public void switchTeam(PlayerWrapper player) {
+        synchronized (teams) {
+            Team oldTeam = teams.remove(player);
+            Team newTeam = oldTeam.getTeamColour() == TeamColour.BLUE ? room.getTeamRed() : room.getTeamBlue();
+
+            teams.put(player, newTeam);
+
+            Player unwrapped = player.unwrap();
+            oldTeam.leave(unwrapped);
+            newTeam.join(unwrapped);
+
+            SwitchTeamMessage message = new SwitchTeamMessage(player.getPlayerId(), newTeam.getTeamColour());
+            players.stream()
+                    .map(PlayerWrapper::getConnection)
+                    .forEach(c -> c.send(message));
+        }
+    }
+
+    public void sendChatMessage(PlayerWrapper player, String message) {
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Document this public method. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://194.135.91.67:9000/coding_rules#rule_key=squid%3AUndocumentedApi)